test/suites/drift/suite_test.go

```diff
@@ -154,4 +146,34 @@ var _ = Describe("Drift", func() {
 		env.EventuallyExpectNotFound(pod, nodeClaim, node)
 		SetDefaultEventuallyTimeout(5 * time.Minute)
 	})
+	It("should mark the nodeclaim as drifted for SubnetDrift if AKSNodeClass subnet id changes", func() {
+		env.ExpectCreated(pod, nodeClass, nodePool)
+
+		By(fmt.Sprintf("expect pod %s to be healthy", pod.Name))
+		env.EventuallyExpectHealthy(pod)
+
+		By("expect created node count to be 1")
+		env.ExpectCreatedNodeCount("==", 1)
+
+		nodeClaim := env.EventuallyExpectCreatedNodeClaimCount("==", 1)[0]
+		node := env.EventuallyExpectNodeCount("==", 1)[0]
+
+		By("triggering subnet drift")
+		// TODO: Introduce azure clients to the tests to get values dynamically and be able to create azure resources inside of tests rather than using a fake id.
+		// this will fail to actually create a new nodeclaim for the drift replacement but should still test that we are marking the nodeclaim as drifted.
+		nodeClass.Spec.VnetSubnetID = lo.ToPtr("/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/sillygeese/providers/Microsoft.Network/virtualNetworks/karpenter/subnets/nodeclassSubnet2")
+		env.ExpectCreatedOrUpdated(nodeClass)
+
+		By(fmt.Sprintf("waiting for nodeclaim %s to be marked as drifted", nodeClaim.Name))
+		env.EventuallyExpectDrifted(nodeClaim)
+		By(fmt.Sprintf("waiting for pod %s to to update", pod.Name))
+		delete(pod.Annotations, corev1beta1.DoNotDisruptAnnotationKey)
+		env.ExpectUpdated(pod)
+
+		By(fmt.Sprintf("expect pod %s, nodeclaim %s, and node %s to eventually not exist", pod.Name, nodeClaim.Name, node.Name))
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		env.EventuallyExpectNotFound(pod, nodeClaim, node)
```

Comment:
Aren't we not going to be able to get a working replacement?

So that means we will still nuke the old pod, nodeClaim, and node that have drifted, even if we don't get a working replacement?

Reply:
see the comment above: `		// this will fail to actually create a new nodeclaim for the drift replacement but should still test that we are marking the nodeclaim as drifted`